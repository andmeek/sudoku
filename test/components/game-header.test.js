import { shallowMount } from '@vue/test-utils'
import GameHeader from '../../src/components/game-header.vue'

describe('GameHeader.vue', () => {
  let wrapper; const game = genTestGame()

  beforeEach(() => {
    wrapper = shallowMount(GameHeader, {
      propsData: {
        game: game
      }
    })
  })

  test('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('emits an exit when the user presses the exit button', () => {
    wrapper.findAll('button').at(0).trigger('click')

    expect(wrapper.emitted().exit).toBeTruthy()
  })
})
