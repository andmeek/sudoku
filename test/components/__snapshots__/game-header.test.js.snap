// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameHeader.vue matches snapshot 1`] = `<div class="game-header"><span class="timer">00:00:00</span> <span class="difficulty">Easy</span> <span><button class="menu inline">Exit Game</button></span></div>`;
